File: ProyectoCivitasRuby/lib/modelo/sorpresa_ir_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
  class SorpresaIrCarcel < Sorpresa
    public_class_method :new
    def initialize(tablero)
      super("Ir a la carcel")
      @tablero = tablero
    end
    
    #@Override
    def aplicarajugador(actual, todos)
      if(jugador_correcto(actual, todos))
        informe(actual,todos)
        todos[actual].encarcelar(@tablero.numcasillacarcel())
      end
    end
    
    #@Override
    def to_s()
      salida = "#{@texto}\nSorpresa para ir a la carcel"
    end
  end
end

File: ProyectoCivitasJava/src/civitas/SorpresaIrCarcel.java
/**
 * Proyecto Civitas
 * Sergio Hervás Cobo
 * Javier Suárez Lloréns
 * Programación y Diseño Orientado a Objetos
 * 2º B3 - Grado en Ingeniería Informática
 * E.T.S.I. Informática y de Telecomunicación
 * Universidad de Granada
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author sergio
 */
public class SorpresaIrCarcel extends Sorpresa {

    private Tablero tablero;

    SorpresaIrCarcel(Tablero tablero) {
        super("Ir a la carcel");
        this.tablero = tablero;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
    }
    
    @Override
    public String toString(){
        String salida =  "Sorpresa para ir a la carcel";
        return salida;
    }
}
